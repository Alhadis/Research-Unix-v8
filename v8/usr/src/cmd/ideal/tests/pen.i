pen
.nf
.IS
box wavy{
    /* perp is a unit vector normal
       to (end - start) */
    var start, end,
        ht, perp, pt1, pt2;
    pt1 = start + (end - start)/4 + ht*perp;
    pt2 = start + 3/4*(end - start) - ht*perp;
    perp = (0,1)*(end - start)/abs(diff);
	var diff;
	diff = end - start;
    conn start to pt1;
    conn pt1 to pt2;
    conn pt2 to end;
    }

box main{
    var a,b,c,d;
    a = (0.0,-4.5);
    b = a + 1;
    c = b + (1,1);
    d = c + 1;
    conn a to b;
    conn b to c using 5*int(abs(b-c)) wavy {ht = 0.05;} <start,end>;
    conn c to d;
    }
.IE
.IS
...forget wavy
.IE
