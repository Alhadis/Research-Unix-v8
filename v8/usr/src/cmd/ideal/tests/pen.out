pen
.nf
.IS
...maxx 3
...maxy -3.5
...minx 0
...miny -4.5
...line 3 -3.5 2 -3.5
...line 2 -3.5 1.98536 -3.58536
...line 1.98536 -3.58536 1.81464 -3.61464
...line 1.81464 -3.61464 1.8 -3.7
...line 1.8 -3.7 1.78536 -3.78536
...line 1.78536 -3.78536 1.61464 -3.81464
...line 1.61464 -3.81464 1.6 -3.9
...line 1.6 -3.9 1.58536 -3.98536
...line 1.58536 -3.98536 1.41464 -4.01464
...line 1.41464 -4.01464 1.4 -4.1
...line 1.4 -4.1 1.38536 -4.18536
...line 1.38536 -4.18536 1.21464 -4.21464
...line 1.21464 -4.21464 1.2 -4.3
...line 1.2 -4.3 1.18536 -4.38536
...line 1.18536 -4.38536 1.01464 -4.41464
...line 1.01464 -4.41464 1 -4.5
...line 1 -4.5 0 -4.5
.IE
.IS
.IE
